client/src/components/popup/PopupModal.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import './PopupModal.scss';
+import XMark from '../../assets/misc/xmark-solid-white.svg';
+
+const PopupModal = ({
+  trigger,
+  setTrigger,
+  heading,
+  children,
+  // bgWidth,
+  // bgHeight,
+  // bgHeightMobile,
+  // containerTop,
+  bodyText,
+  exitIcon,
+  blurBackground,
+}) => {
+  const [hasBeenOpened, setHasBeenOpened] = useState(false);
+
+  if (hasBeenOpened === false && trigger === false) {
+    return <div></div>;
+  } else if (hasBeenOpened === false && trigger === true) {
+    setHasBeenOpened(true);
+  }
```

Comment:
I would wrap this in a `useEffect` - React can sometimes be a little confusing otherwise with when it re-renders the components